I just added the menu

ozanmujde opened the pull request:
I added the Menu feature with using java awt

ozanmujde committed:
I just add menu
It uses java awt and for restart the game it use gradle -p command  on command line

cagdasgerede commented:
Bu kadar mı yapacağın değişiklikler?

Eğer öyleyse bir demo videosu da ekle buraya.

ozanmujde commented:
Hocam 3 buton istenmişti onları ekledim ama eğer başka bir şey eklememi isterseniz ekleyebilirim demo videosundan önce.

cagdasgerede commented:
Daha bir kaç kez daha git gel olacak burada. O yüzden şu ana kadar yaptıklarını demo et. Sonra yeniden video çekmen gerekecek.

ozanmujde committed:
remove gradle build when restarting

ozanmujde commented:
https://youtu.be/7EfboHxYHdU son commiti çok küçük bir değişiklik diye buraya yüklemedim,  başka değişikliklerle eklerim diye düşündüm.

ozanmujde commented:
Hocam merge edeyim mi kodu ?

ozanmujde commented:
Hocam  bir kaç eklenti daha yapmak istiyorum ama onları finaller 17 sinde bitiyor ondan sonra yapsam sıkıntı olur mu?

cagdasgerede commented on src/main/java/pl/nogacz/checkers/application/Menu.java:
```diff
@@ -0,0 +1,72 @@
+    public static void restartApplication()
+    {
+        try {
+            final String javaBin = System.getProperty("java.home") + File.separator + "bin" + File.separator + "java";
```
Bu şekilde olmaz. Programı kapatıp yeniden çalıştırmadan bu işi yapman lazım.

ozanmujde replied:
Hocam bunu 17 sinden sonra yapsam olur mu? finaller geldide.

cagdasgerede commented on src/main/java/pl/nogacz/checkers/application/Menu.java:
```diff
@@ -0,0 +1,72 @@
+    private JButton NewGame_button = new JButton("New Game");
+    private JButton Exit_button = new JButton("EXIT");
+
+    public Menu() {
```
Burada yaptığın arayüz programın temasıyla uyuşmuyor. Görsel olarak oyun tahtası ile daha uyumlu bir arayüz bekliyorum. En azından Javanın default mavi kocaman kare düğmelerinden daha özenli bir arayüz beklentim var.

ozanmujde replied:
Done

cagdasgerede reviewed (changes requested):
Needs more work. See my comments.

ozanmujde reviewed (commented):
Hocam pardon yanlışlıkla yorumunuzu sildim
<<<Bu şekilde olmaz.Programı kapatıp yeniden çalıştırmadan bu işi yapman lazım.>>

ozanmujde commented:
Hocam pazartesi perşembe arası final haftasıda perşembe dediğiniz değişiklilere başlasam sıkıntı yaratır mı?

cagdasgerede commented:
Ozan Perşembe'ye kadar bekleyemeyiz.

ozanmujde committed:
Removed Menu and add new Inner Class
I do that due to private and static problems in the Board class
Also now ı stop the game while resume is pressed
Now ı do not restart the game, now I remove all pawns and add
them again

ozanmujde committed:
Add starter menu

ozanmujde committed:
I use button maker to do buttons look better

ozanmujde commented:
Hocam söylediğiniz şeyleri değiştirdim videoyuda atıyorum şimdi kod açıklamasıyla

ozanmujde commented:
https://youtu.be/q-RcErX6Clg yeni videom yayında.

cagdasgerede commented on src/main/java/pl/nogacz/checkers/application/StartMenu.java:
```diff
@@ -0,0 +1,57 @@
+        start_board = board;  
+        board.isMenuActive=true;
+        setSize(WIDTH,HEIGHT);
+        setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE); // dont need that but maybe someone want to remove undecorated
```
gereksiz yorum

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/application/StartMenu.java:
```diff
@@ -0,0 +1,57 @@
+    public void actionPerformed(ActionEvent actionEvent) {
+        String command = actionEvent.getActionCommand();
+        if(command.equals("Start")) {
+                start_board.isMenuActive=false;
```
fix indentation

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/application/StartMenu.java:
```diff
@@ -0,0 +1,57 @@
+    public void actionPerformed(ActionEvent actionEvent) {
+        String command = actionEvent.getActionCommand();
+        if(command.equals("Start")) {
+                start_board.isMenuActive=false;
```
fix spacing

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/application/StartMenu.java:
```diff
@@ -0,0 +1,57 @@
+        String command = actionEvent.getActionCommand();
+        if(command.equals("Start")) {
+                start_board.isMenuActive=false;
+                this.dispatchEvent(new WindowEvent(this,WindowEvent.WINDOW_CLOSING));
```
fix spacing

`this,Windo` => `this, Window`

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -6,14 +6,25 @@
 import javafx.scene.input.KeyCode;
 import javafx.scene.input.KeyEvent;
 import javafx.scene.input.MouseEvent;
+import jdk.javadoc.internal.tool.Start;
```
import sırası yanlış

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -6,14 +6,25 @@
 import pl.nogacz.checkers.pawns.PawnColor;
 import pl.nogacz.checkers.pawns.PawnMoves;
 
+import javax.swing.*;
```
import sırası yanlış

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -35,7 +46,7 @@
     private boolean isGameEnd = false;
     private int roundWithoutKick = 0;
-
+    public boolean isMenuActive=false;
```
fix spacing

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -35,7 +46,7 @@
     private boolean isGameEnd = false;
     private int roundWithoutKick = 0;
-
+    public boolean isMenuActive=false;
```
Do not make this public. Instead implement an accessor method.

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -77,9 +88,13 @@ private void addStartPawn() {
         for(Map.Entry<Coordinates, PawnClass> entry : board.entrySet()) {
             Design.addPawn(entry.getKey(), entry.getValue());
         }
+        
```
gereksiz newline

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -404,4 +425,109 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
+            //Initialize
+            super("Game Menu");
+            setSize(WIDTH,HEIGHT);
+            setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE); // dont need that but maybe someone want to remove undecorated
```
gereksiz yorum

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -404,4 +425,109 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
+            setSize(WIDTH,HEIGHT);
+            setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE); // dont need that but maybe someone want to remove undecorated
+            this.setUndecorated(true);
+            this.setBackground(new Color(0, 0, 0,100));
```
fix spacing.

Bu arada bu ne renk oluyor?

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -404,4 +425,109 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
+            this.add(Resume_button);
+            this.add(NewGame_button);
+            this.add(Exit_button);
+            //Set Bounds
```
gereksiz yorum. Zaten hemen izleyen kod setbounds yapmıyor.

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -404,4 +425,109 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
+            Resume_button.setBounds(150, 50, 225, 80);
+            NewGame_button.setBounds(150, 200, 280, 80);
+            Exit_button.setBounds(150, 350, 150, 80);
+            //Give them Func
```
Anlamsız yorum. Kaldırılabilir.

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -404,4 +425,109 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
+            String exit_path = this.getClass().getClassLoader().getResource("button_exit.png").getFile();
+            String resume_path = this.getClass().getClassLoader().getResource("button_resume.png").getFile();
+            String new_game_path = this.getClass().getClassLoader().getResource("button_new-game.png").getFile();
+            Exit_button.setIcon(new ImageIcon(exit_path));
```
değişken isimlendirmesi consistent değil. Bunların isimleri küçük harfle başlamalı

Exit_button -> exit_button
Resume_button -> resume_button
...
vs.

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -404,4 +425,109 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
+        @Override
+        public void actionPerformed(ActionEvent actionEvent) {
+            String command = actionEvent.getActionCommand();
+            System.out.println("hey");
```
sysout görmek istemiyorum

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -404,4 +425,109 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
+            String command = actionEvent.getActionCommand();
+            System.out.println("hey");
+            if(command.equals("Resume")) {
+                System.out.println("pressed");
```
ditto

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -404,4 +425,109 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
+            System.out.println("hey");
+            if(command.equals("Resume")) {
+                System.out.println("pressed");
+                    isMenuActive=false;
```
fix spacing

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -404,4 +425,109 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
+                    this.dispatchEvent(new WindowEvent(this,WindowEvent.WINDOW_CLOSING));
+            }else if(command.equals("New Game")) {
+                restart();
+            }else if(command.equals("EXIT")){
```
Bu String değerlerini enum olarak tanımla ve o şekilde kod içerisinde kullan.

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -404,4 +425,109 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
+        public void restart()
+        {
+            Platform.runLater(() -> { 
+                for(Map.Entry<Coordinates, PawnClass> entry : board.entrySet()) {
```
`for` içerde fakat geri kalanlar yanlış aling edilmiş. Alingmentı düzelt.

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -404,4 +425,109 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
+                 }  
+            board.clear(); 
+
+            board.put(new Coordinates(1, 0), new PawnClass(Pawn.PAWN, PawnColor.BLACK));
```
Burayı bir döngü olarak yaz.

for i = 1, 3, 5, 7; j = 0, 2
  board.put(...)

for i = 0, 2, 4, 6; j = 1
  board.put()

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -404,4 +425,109 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
+
+            });
+
+           
```
gereksiz newline karakterleri

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -404,4 +425,109 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
+            this.add(Exit_button);
+            //Set Bounds
+            String exit_path = this.getClass().getClassLoader().getResource("button_exit.png").getFile();
+            String resume_path = this.getClass().getClassLoader().getResource("button_resume.png").getFile();
```
Burada tekrar eden kodlar var. Bu kodu bir metot içerisine al ve her bir değişken için o metodu çağır. Kodu sadeleşir okuması anlaması kolaylaşır.
```
getImageIconForFileName(String name) {
  File f = this.getClass().getClassLoader().getResource(name).getFile();
  return new ImageIcon(f);
}

newGameButton.setIcon(getImageIconForFileName("button_new-game.png");

```

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -404,4 +425,109 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
+            Exit_button.setIcon(new ImageIcon(exit_path));
+            Resume_button.setIcon(new ImageIcon(resume_path));
+            NewGame_button.setIcon(new ImageIcon(new_game_path));
+            Resume_button.setBounds(150, 50, 225, 80);
```
Burada değişken isimleri consistent değil.

Ya camelCase kullan ya da underscore. Ama sen ikisini de karıştırmışsın.

Kodun geneline baktığımızda da bence hepsini camelCase'e çevirmek en doğrusu.

`NewGame_button` => `newGameButton`

Tüm değişikliklerine uygula bu dediğimi.

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/application/StartMenu.java:
```diff
@@ -0,0 +1,57 @@
+            System.exit(0);
+        }
+    }
+} 
```
newline

ozanmujde replied:
Done

cagdasgerede reviewed (changes requested):
Kod içerisinde bıraktığım notlara göre kodunu güncelle.

Ayrıca kodunda herhangi bir test yok. Yeni eklediğin davranışlar ile ilgili testler ekle.

ozanmujde committed:
getFileMethod and Command enum is added

ozanmujde commented:
> Kod içerisinde bıraktığım notlara göre kodunu güncelle.
> 
> Ayrıca kodunda herhangi bir test yok. Yeni eklediğin davranışlar ile ilgili testler ekle.

Değişiklileri yaptım 1 2 yerde ekstra space errora sebep oldum ama onun için commit atmak istemedim. Test caseleri ekleyeceğim hocam. bu arada yeni eklememi istediğiniz bir özellik ya da değişmesini istediğiniz bir şey var mı?
Saygılarımla.

ozanmujde commented:
Hocam test bizim dosyada otomatik olarak yok.Diğer arkadaşlarla çelişmemesi için belli bir formatta mı olmalı?

ozanmujde committed:
Add dependencies for start testing

ozanmujde commented:
Hocam bir de benim eklediğim özellik ana kodun private datasını kullanan bir method private değişkenler üzerinde test case i tam olarak nasıl yazmam gerekiyor? Bunların dışınca tek bir classın bir methodunu dependencies hatası yüzünden test edemiyorun çok test yazmak gerekiyorda test işini sonra yapıp önce eklentileri eklesem süre dolayısıyla

ozanmujde committed:
I add test case for StartMenu

ozanmujde committed:
Undo ui tests

ozanmujde commented:
`(base) [ozan@ozan-20hns03t00 JavaCheckers]$ gradle test

> Configure project :
Project : => no module-info.java found

> Task :test
java.lang.instrument.IllegalClassFormatException: Error while instrumenting sun/util/resources/cldr/provider/CLDRLocaleDataMetaInfo.
        at org.jacoco.agent.rt.internal_43f5073.CoverageTransformer.transform(CoverageTransformer.java:94)
        at java.instrument/java.lang.instrument.ClassFileTransformer.transform(ClassFileTransformer.java:246)
        at java.instrument/sun.instrument.TransformerManager.transform(TransformerManager.java:188)
        at java.instrument/sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:563)
        at java.base/java.lang.ClassLoader.defineClass2(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1108)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:183)
        at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:784)
        at java.base/jdk.internal.loader.BuiltinClassLoader.findClassInModuleOrNull(BuiltinClassLoader.java:705)
        at java.base/jdk.internal.loader.BuiltinClassLoader.findClass(BuiltinClassLoader.java:586)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:634)
        at java.base/java.lang.Class.forName(Class.java:546)
        at java.base/java.util.ServiceLoader.loadProvider(ServiceLoader.java:854)
        at java.base/java.util.ServiceLoader$ModuleServicesLookupIterator.hasNext(ServiceLoader.java:1078)
        at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1301)
        at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1386)
        at java.base/sun.util.cldr.CLDRLocaleProviderAdapter$1.run(CLDRLocaleProviderAdapter.java:89)
        at java.base/sun.util.cldr.CLDRLocaleProviderAdapter$1.run(CLDRLocaleProviderAdapter.java:86)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:554)
        at java.base/sun.util.cldr.CLDRLocaleProviderAdapter.<init>(CLDRLocaleProviderAdapter.java:86)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:64)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
        at java.base/sun.util.locale.provider.LocaleProviderAdapter.forType(LocaleProviderAdapter.java:188)
        at java.base/sun.util.locale.provider.LocaleProviderAdapter.findAdapter(LocaleProviderAdapter.java:287)
        at java.base/sun.util.locale.provider.LocaleProviderAdapter.getAdapter(LocaleProviderAdapter.java:258)
        at java.base/java.util.Calendar.createCalendar(Calendar.java:1693)
        at java.base/java.util.Calendar.getInstance(Calendar.java:1661)
        at java.base/java.text.SimpleDateFormat.initializeCalendar(SimpleDateFormat.java:677)
        at java.base/java.text.SimpleDateFormat.<init>(SimpleDateFormat.java:621)
        at java.base/java.text.SimpleDateFormat.<init>(SimpleDateFormat.java:600)
        at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.getLastResortErrorLogFile(SystemApplicationClassLoaderWorker.java:162)
        at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:111)
        at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
        at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
        at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
Caused by: java.io.IOException: Error while instrumenting sun/util/resources/cldr/provider/CLDRLocaleDataMetaInfo.
        at org.jacoco.agent.rt.internal_43f5073.core.instr.Instrumenter.instrumentError(Instrumenter.java:159)
        at org.jacoco.agent.rt.internal_43f5073.core.instr.Instrumenter.instrument(Instrumenter.java:109)
        at org.jacoco.agent.rt.internal_43f5073.CoverageTransformer.transform(CoverageTransformer.java:92)
        ... 37 more
Caused by: java.lang.IllegalArgumentException: Unsupported class file major version 59
        at org.jacoco.agent.rt.internal_43f5073.asm.ClassReader.<init>(ClassReader.java:195)
        at org.jacoco.agent.rt.internal_43f5073.asm.ClassReader.<init>(ClassReader.java:176)
        at org.jacoco.agent.rt.internal_43f5073.asm.ClassReader.<init>(ClassReader.java:162)
        at org.jacoco.agent.rt.internal_43f5073.core.internal.instr.InstrSupport.classReaderFor(InstrSupport.java:280)
        at org.jacoco.agent.rt.internal_43f5073.core.instr.Instrumenter.instrument(Instrumenter.java:75)
        at org.jacoco.agent.rt.internal_43f5073.core.instr.Instrumenter.instrument(Instrumenter.java:107)
        ... 38 more

BoardTest > trying() FAILED
    java.lang.ExceptionInInitializerError at BoardTest.java:51
        Caused by: java.lang.RuntimeException at BoardTest.java:51

1 test completed, 1 failed

> Task :test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/ozan/Desktop/Lectures/BIL_481/JavaCheckers/JavaCheckers/build/reports/tests/test/index.html

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.7/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 2s
4 actionable tasks: 1 executed, 3 up-to-date`

ozanmujde committed:
latest tries for test but nothing changed

ozanmujde commented:
Son attığım kod bu hatayı veriyor  hocam gerçekten uğraşıp çözemediğim için soruyorum daha fazla uğraşırsam yetişemyecek gibi.Her yerde static kullanılmış staticlerde test çalışmıyor hocam internette bir cevap bulamadım.En son kendim restart için ayrı class açıp deneyeyim dedim ama static şeylerde direkt hata atıyor methodlar daha teste çağırılırken hata atıyor hatta Design'i mocklamaya çalışıtığımda bile hata alıyorum

ozanmujde committed:
Test for restart Method are added

We cannot mock or use Design.class or any JavaFx application so I could not use
or test its properties and they were in the code before I write mines so I dont their properties
also I did not write ui test just unit test for my restart method

ozanmujde commented:
Hocam javafx kullanan design,board gibi methodlara test yazılmıyordu.O yüzden restart methoda ayrı class açtım ona test yazdım.Yazılan EndGame classı pek iyi çalışmıyordu zamanla ona da evrilebilir.Ama test etmek için yazılan bir class aslında.Test için de ayrı bir method yazdım içinde design olmayan ama aynı method design testte toolbox exception atıyordu dependencies den dolayı çözmek hem sıkıntıydı.removePawn zaten adamın yazdığı bir method olduğundan onları test etmek gerek yoktur diye düşündüm.

cagdasgerede commented on src/test/java/pl/nogacz/checkers/application/RestartTest.java:
```diff
@@ -0,0 +1,77 @@
+package pl.nogacz.checkers.application;
+
+import pl.nogacz.checkers.application.Restart;
+import pl.nogacz.checkers.board.*;
```
yıldız kullanma

ozanmujde replied:
Done

cagdasgerede commented on src/test/java/pl/nogacz/checkers/application/RestartTest.java:
```diff
@@ -0,0 +1,77 @@
+import pl.nogacz.checkers.pawns.PawnClass;
+import pl.nogacz.checkers.pawns.PawnColor;
+import pl.nogacz.checkers.pawns.PawnMoves;
+import static org.mockito.Mockito.*;
```
yıldız kullanma

ozanmujde replied:
Mockito silindi.
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -401,7 +424,80 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
+        //Add your Command names here 
+        //For use them as string call Commands.EXIT.name() or for another class
+        //Board.Commands.Resume.toString()
+        Resume,
```
Bazıları hep büyük harf, bazıları ise normal, neden?
Consistent değil. Hepsini all capital yap.

ozanmujde replied:
NEW_GAME yaptım.İsterseniz değiştirebilirim

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -401,7 +424,80 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
+            if(command.equalsIgnoreCase(Commands.Resume.name())) {
+                isMenuActive=false;
+                this.dispatchEvent(new WindowEvent(this,WindowEvent.WINDOW_CLOSING));
+            }else if(command.equalsIgnoreCase(Commands.NewGame.name())) {
```
`} else if` bir boşluk bırak parentez kapamalardan sonra. Aşağıdaki else te de aynı durum var.

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/board/Board.java:
```diff
@@ -1,19 +1,33 @@
 
+import javax.swing.*;
+
+import java.awt.*;
```
yıldız kullanma, kullandıklarını tek tek sırala

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/application/StartMenu.java:
```diff
@@ -0,0 +1,87 @@
+            menuOpen = false;
+            this.dispatchEvent(new WindowEvent(this, WindowEvent.WINDOW_CLOSING));
+        }else if(command.equalsIgnoreCase(Board.Commands.EXIT.name())) {
+            System.exit(1735);//random number for not match with an error
```
Anlamadım: "//random number for not match with an error"

ozanmujde replied:
Hocam eğer ordaki exit in testini yapmak istersek hatadan dolayı değilde elle exit yapıldığı anlaşılsın diye koymuştum.

ozanmujde replied:
isterseniz gidebilir

cagdasgerede commented on src/main/java/pl/nogacz/checkers/application/StartMenu.java:
```diff
@@ -0,0 +1,87 @@
+        setSize(WIDTH, HEIGHT);
+        setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+        this.setUndecorated(true);
+        Board ozan = new Board(); // is working
```
ozan? board ile değiştir.

commenti de sil.

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/application/StartMenu.java:
```diff
@@ -0,0 +1,87 @@
+import jdk.javadoc.internal.doclets.formats.html.SourceToHTMLConverter;
+import jdk.javadoc.internal.doclets.toolkit.taglets.SystemPropertyTaglet;
+
+import java.awt.event.*;
```
yıldız kullanma

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/application/StartMenu.java:
```diff
@@ -0,0 +1,87 @@
+import java.awt.Color;
+
+import javax.sound.midi.Soundbank;
+import javax.swing.*;
```
yıldız kullanma

ozanmujde replied:
Done

cagdasgerede commented on src/main/java/pl/nogacz/checkers/application/StartMenu.java:
```diff
@@ -0,0 +1,87 @@
+import javax.sound.midi.Soundbank;
+import javax.swing.*;
+
+import jdk.javadoc.internal.doclets.formats.html.SourceToHTMLConverter;
```
Burada kullanılmayan importlar var

ozanmujde replied:
Done

cagdasgerede reviewed (changes requested):
Kodlarını bıraktığım notlara göre güncelle.

Jacoco'yu çalıştır ve aşağıda listelediğim dosyaların coverage durumunu gösteren ekran görüntüsünü PRda paylaş:
- Restart.java

Bir de yorumlara cevap verip Resolve Converation düğmesine basmayı unutma.

Review etmemi istediğinde bana assign etmeyi unutma.

ozanmujde commented:
![Screenshot_2020-12-21 Restart](https://user-images.githubusercontent.com/46843218/102765401-9f7ab180-438d-11eb-82c3-02d9b81d4823.png)

ozanmujde committed:
small refactoring

ozanmujde commented:
Hocam yıldızları ne intellij idea ne de visual studio code işlevini yapmadığı için elle yapıyorum.O yüzden o biraz sürebilir document okuyarak vs.Hemen hızlıca bunları atmak istedim Desing classını test ederken mock() da dahi hata atıyordu o yüzden kalan yerlerin testini yazmıştım.Eğer Design'ı da isterseniz biraz mock'u okuyarak vs deneyebilirim

cagdasgerede reviewed (changes requested):
Tekrar sana assign ediyorum

ozanmujde committed:
small changes on import

ozanmujde committed:
Little func. update
Now user can only open game menu once. No matter how much ze pressed
and also now  N and R keys working which we can use them for restart
the game

ozanmujde commented:
Başka bir şey eklememi denememi istiyorsanız yapabilirim hocam.
